autosklearn/ensemble_building/manager.py

```diff
@@ -0,0 +1,371 @@
+    def __init__(
+        self,
+        backend: Backend,
+        dataset_name: str,
+        task: int,
+        metric: Scorer,
+        time_left_for_ensembles: float = 10,
+        max_iterations: int | None = None,
+        pynisher_context: str = "fork",
+        ensemble_size: int = 10,
+        ensemble_nbest: int | float = 100,
+        max_models_on_disc: int | float | None = None,
+        seed: int = 1,
+        precision: int = 32,
+        memory_limit: int | None = None,
+        read_at_most: int | None = None,
+        logger_port: int = logging.handlers.DEFAULT_TCP_LOGGING_PORT,
+        random_state: int | np.random.RandomState | None = None,
+        start_time: float | None = None,
+    ):
+        """SMAC callback to handle ensemble building
+
+        Parameters
+        ----------
+        backend: Backend
+            backend to write and read files
+
+        dataset_name: str
+            name of dataset
+
+        task: int
+            Type of ML task
+
+        metric: Scorer
+            Metric to compute the loss of the given predictions
+
+        time_left_for_ensemble: float = 10
+            How much time is left for the task in seconds.
+            Job should finish within this allocated time
+
+        max_iterations: int | None = None
+            maximal number of iterations to run this script. None indicates no limit
+            on iterations.
+
+        pynisher_context: "spawn" | "fork" | "forkserver" = "fork"
+            The multiprocessing context for pynisher.
+
+        ensemble_size: int = 10
+            maximal size of ensemble
+
+        ensemble_nbest: int | float = 100
+            If int: consider only the n best prediction
+            If float: consider only this fraction of the best models
+            If performance_range_threshold > 0, might return less models
+
+        max_models_on_disc: int | float | None = None
+           Defines the maximum number of models that are kept in the disc.
+
+           If int, it must be greater or equal than 1, and dictates the max
+           number of models to keep.
+
+           If float, it will be interpreted as the max megabytes allowed of
+           disc space. That is, if the number of ensemble candidates require more
+           disc space than this float value, the worst models will be deleted to
+           keep within this budget. Models and predictions of the worst-performing
+           models will be deleted then.
+
+           If None, the feature is disabled. It defines an upper bound on the
+           models that can be used in the ensemble.
+
+        seed: int = 1
+            Seed used for the inidividual runs
+
+        precision: 16 | 32 | 64 | 128 = 32
+            Precision of floats to read the predictions
+
+        memory_limit: int | None = None
+            Memory limit in mb. If ``None``, no memory limit is enforced.
+
+        read_at_most: int | None = 5
```

Comment:
It is, the documentation should be updated. It used to have no default but always be set to `np.inf` in `automl.py`. I changed it to `None` to indicate no limit as `np.inf` would cause problems if using it as an index.

Will update documentation